src/libsyntax/parse/parser.rs

```diff
@@ -4440,14 +4440,20 @@ impl<'a> Parser<'a> {
                     let bounded_lifetime =
                         self.parse_lifetime()?;
 
-                    self.eat(&token::Colon);
+                    self.expect(&token::Colon)?;
 
                     let bounds =
                         self.parse_lifetimes(token::BinOp(token::Plus))?;
 
                     let hi = self.prev_span.hi;
                     let span = mk_sp(lo, hi);
 
+                    if bounds.is_empty() {
+                        self.span_err(span,
+                                      "each predicate in a `where` clause must have \
+                                       at least one bound in it");
+                    }
```

Comment:
I think we shouldn't make this an error until we reach a decision empty bounds lists in #37511.